gears/src/x/ante.rs

```diff
@@ -450,6 +451,60 @@ impl<
                         account_number: acct.get_account_number(),
                     }
                     .encode_to_vec(),
+                    SignMode::LegacyAminoJson => {
+                        let mut msgs = vec![];
+                        for msg in tx.tx.get_msgs() {
+                            // TODO: don't have proper value getter, maybe extend trait
+                            let mut value: serde_json::Map<String, serde_json::Value> =
+                                serde_json::from_slice(
+                                    &serde_json::to_vec(&msg)
+                                        .map_err(|_| AnteError::LegacyAminoJson)?,
+                                )
+                                .map_err(|_| AnteError::LegacyAminoJson)?;
+                            value.remove("@type");
+
+                            msgs.push(std_sign_doc::Msg {
+                                kind: std_sign_doc::proto_type_url_to_legacy_amino_type_url(
+                                    msg.type_url(),
+                                ),
+                                value,
+                            })
+                        }
+                        let amount = {
+                            let fee = tx
+                                .tx
+                                .auth_info
+                                .fee
+                                .amount
+                                .clone()
+                                .ok_or(AnteError::MissingFee)?;
+                            fee.into_inner()
+                                .into_iter()
+                                .map(|coin| std_sign_doc::Coin {
+                                    amount: coin.amount.to_string(),
+                                    denom: coin.denom.to_string(),
+                                })
+                                .collect()
+                        };
+                        let doc = std_sign_doc::StdSignDoc {
+                            account_number: acct.get_account_number().to_string(),
+                            chain_id: ctx.chain_id().to_string(),
+                            fee: std_sign_doc::StdFee {
+                                amount,
```

Comment:
Is it possible to implement `From<GearsFee> for StdFee`

Reply:
I'll add.